config/configmodels/configmodels.go

```diff
@@ -221,3 +253,24 @@ func (ext *ExtensionSettings) Type() Type {
 }
 
 var _ Extension = (*ExtensionSettings)(nil)
+
+// DecodeTypeAndName decodes the type and name from a full name "type/name" format.
+func DecodeTypeAndName(fullName string) (string, string, error) {
+	items := strings.SplitN(fullName, typeAndNameSeparator, 2)
+	if len(items) == 0 || items[0] == "" {
+		return "", "", errors.New("type/name format must have the type part")
+	}
+
+	var nameVal string
+	if len(items) > 1 {
+		// "name" part is present.
+		nameVal = strings.TrimSpace(items[1])
```

Comment:
I wonder if we truly need to trim space, shouldn't we fail if there is a space after `/`?

Reply:
We can discuss this in a separate issue/pr. Not combining multiple changes. This is what we did in config.go